app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala

```diff
@@ -85,15 +84,61 @@ object PayePersonalDetails {
         def readDob(): JsResult[LocalDate] =
           readData[LocalDate](List(read("dateOfBirth")), "DateOfBirth")
 
-        def readAddress(): JsResult[Address] =
-          readData[Address](List(readSeq("addresses", "2"), readSeq("addresses", "1")), "Address") // 1–Residential Address, 2–Correspondence Address
+        def readAddress(): JsResult[Address] = {
+          List(readSeq("addresses", "2"), readSeq("addresses", "1")) //7–Mobile Telephone Number, 1–Daytime Home Telephone Number
+            .find(_.isDefined)
+            .fold[JsResult[Address]](
+              JsError("PayePersonalDetails : could not retrieve Address for user")
+            ) { x ⇒
+                x.fold[JsResult[Address]](
+                  JsError("PayePersonalDetails : could not retrieve Address for user")
+                ) { v ⇒
+                    val line1 = (v \ "line1").as[String]
+                    val line2 = (v \ "line2").as[String]
+                    val line3 = (v \ "line3").asOpt[String]
+                    val line4 = (v \ "line4").asOpt[String]
+                    val line5 = (v \ "line5").asOpt[String]
+                    val postcode = (v \ "postcode").as[String]
+                    val countryCode = (v \ "countryCode").as[Int]
+
+                    JsSuccess(Address(line1, line2, line3, line4, line5, postcode, CountryCode.getCodeFor(countryCode.toString)))
+                  }
+              }
+        }
+
+        def readPhoneNumber(): JsResult[Option[String]] = {
+          List(readSeq("phoneNumbers", "7"), readSeq("phoneNumbers", "1")) //7–Mobile Telephone Number, 1–Daytime Home Telephone Number
+            .find(_.isDefined)
+            .fold[JsResult[Option[String]]](
+              JsSuccess(None)
+            ) { x ⇒
+                x.fold[JsResult[Option[String]]](
+                  JsSuccess(None)
+                ) { v ⇒
+                    val callingCode = (v \ "callingCode").asOpt[Int]
+                    val convertedAreaDiallingCode = (v \ "convertedAreaDiallingCode").asOpt[String]
+                    val telephoneNumber = (v \ "telephoneNumber").asOpt[String]
+
+                    (callingCode, convertedAreaDiallingCode, telephoneNumber) match {
+                      case (Some(ccKey), Some(cadc), Some(t)) ⇒
+                        val cc = CallingCodes.getCodeFor(ccKey).getOrElse("")
+                        JsSuccess(Some(s"+$cc${cadc.replaceFirst("^0+(?!$)", "")}$t"))
```

Comment:
`replaceFirst` replaces all the leading `zeros` in the string - currently this fails for `250 - 0` which has only one `0` and it will be replaced by `""`